app/uk/gov/hmrc/tai/viewModels/TaxCodeViewModel.scala

```diff
@@ -25,25 +27,39 @@ import uk.gov.hmrc.tai.viewModels.TaxCodeDescriptor
 case class TaxCodeViewModel(title: String,
                             mainHeading: String,
                             ledeMessage: String,
-                            taxCodeDetails: Seq[DescriptionListViewModel])
+                            taxCodeDetails: Seq[DescriptionListViewModel],
+                            preHeader: String)
 
 object TaxCodeViewModel extends ViewModelHelper with TaxCodeDescriptor {
 
-  def apply(taxCodeIncomes: Seq[TaxCodeIncome], scottishTaxRateBands: Map[String, BigDecimal])(implicit messages: Messages): TaxCodeViewModel = {
+  def apply(taxCodeIncomes: Seq[TaxCodeIncome], scottishTaxRateBands: Map[String, BigDecimal], year: TaxYear = TaxYear())(implicit messages: Messages): TaxCodeViewModel = {
+
+    val previousOrCurrent = if (year <= TaxYear().prev) ".prev" else ""
```

Comment:
Non blocking comment but the previousOrCurrent logic is in two places currently. 

We could move all of it to something that actually understands message tenses. This may be a new thing that is a layer between messages or much simpler where it just has some helper methods that help understand tenses.

This would hopefully help make the code more reusable